### PR TITLE
Df loss

### PR DESCRIPTION
**Pull request by FelixMoeller3**
Implementing Two-View Training with jpldd.

- Sparse NRE loss to train points and descriptors
- Consistency loss for Distancefield
- We integrate both into n new 2 vie pipeline so one view and two-view losses can both be used

### CODE REVIEW

File: gluefactory/models/extractors/joint_point_line_extractor.py [hunk 1]
```diff
@@ -774,3 +795,44 @@ def _get_warped_outputs(self, data):
         with torch.no_grad():
             warped_outputs = self({"image": warped_imgs})
         return warped_outputs, Hs
+    
+    import torch
+import torch.nn.functional as F
+
+def compute_matches(keypoints_im1: torch.Tensor, keypoints_im2: torch.Tensor, H: torch.Tensor) -> torch.Tensor:
```
Comment (hardik01shah) [inline gluefactory/models/extractors/joint_point_line_extractor.py:802]: From the paper: I think a groudtruth match is a two-way match. 
We should also warp keypoints_im1 using H, calculate the distance and take a mean. Then we check for points with pairwise distances less than 3.

File: gluefactory/models/extractors/joint_point_line_extractor.py [hunk 2]
```diff
@@ -774,3 +795,44 @@ def _get_warped_outputs(self, data):
...
+    similarity_matrix = torch.matmul(desc1, desc2.t())  # (M x M)
+
+    # Apply temperature scaling
+    similarity_matrix /= temperature
+
+    # Create ground truth labels
+    labels = torch.arange(len(matches), device=descriptors1.device)
+
+    # Compute cross-entropy loss
+    loss = F.cross_entropy(similarity_matrix, labels)
```
Comment (hardik01shah) [inline gluefactory/models/extractors/joint_point_line_extractor.py:835]: I think calculating the loss as:
`loss = -torch.log_softmax((similarity - 1 )/ temperature)` 
ensures correctness and consistency with the paper.

File: gluefactory/models/extractors/joint_point_line_extractor.py (hunk 1 above)
Comment (hardik01shah) [reply gluefactory/models/extractors/joint_point_line_extractor.py:802]: Page 7: Section V C. of https://arxiv.org/pdf/2304.03608
For the training code, they use it from ALIKE which is available at https://github.com/Shiaoming/ALIKE/raw/main/assets/ALIKE_code.zip

File: gluefactory/models/extractors/joint_point_line_extractor.py (hunk 2 above)
Comment (RaphaelKreft) [reply gluefactory/models/extractors/joint_point_line_extractor.py:835]: As F.cross_entropy() already performs softmax and the temperature is already divided. But I agree we should subtract 1 before.

File: gluefactory/models/extractors/joint_point_line_extractor.py (hunk 1 above)
Comment (RaphaelKreft) [reply gluefactory/models/extractors/joint_point_line_extractor.py:802]: So you think we should use mutual nearest neighbor matching? From ALIKED we thought one point from Img A could be matched to multiple points in img B
